framework/aster-frame/src/vm/dma/dma_coherent.rs

```diff
@@ -55,6 +55,7 @@ impl DmaCoherent {
         }
         if !is_cache_coherent {
             let mut page_table = KERNEL_PAGE_TABLE.get().unwrap().lock();
+            start_paddr.checked_add(frame_count * PAGE_SIZE).unwrap();
```

Comment:
Add a comment to make the intention more clear: the intention is not to do one checked addition, but to prevent integer overflow for all possible addresses later used in this function.

```rust
            // Ensure that the addresses used later will not overflow 
            start_paddr.checked_add(frame_count * PAGE_SIZE).unwrap();
```

And this line should be moved to the beginning of this function to avoid repeating it in multiple branches.

Reply:
Great suggestions. I have moved the check to the beginning now.